### PR TITLE
feat: add vulnerabilityreports controller

### PR DESCRIPTION
**Pull request by zreigz**

### CODE REVIEW

File: internal/controller/vulnerabilityreports_controller.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+			PackageType:      lo.ToPtr(v.PackageType),
+			PkgPath:          lo.ToPtr(v.PkgPath),
+			PublishedDate:    lo.ToPtr(v.PublishedDate),
+			LastModifiedDate: lo.ToPtr(v.LastModifiedDate),
+		}
+		vulnerabilityAttributes = append(vulnerabilityAttributes, vulnerabilityAttr)
+	}
+
+	return &console.VulnerabilityReportAttributes{
+		ArtifactURL:     nil,
```
Comment (zreigz) [inline internal/controller/vulnerabilityreports_controller.go:119]: @michaeljguarino I'm not sure how to get this.

Comment (michaeljguarino) [reply internal/controller/vulnerabilityreports_controller.go:119]: you would get this by concatenating registry + repository + tag/digest, in the usual docker image url format.  Basically want it to be easily accessible for the ui, and also to potentially cross-ref elsewhere

**Comment by zreigz** [comment]
@michaeljguarino  everything seems to work fine PTAL

File: internal/controller/vulnerabilityreports_controller.go [hunk 2]
```diff
@@ -152,6 +167,23 @@ func (r *VulnerabilityReportReconciler) getObjectFromOwnerReference(ctx context.
 }
 
 func (r *VulnerabilityReportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	logger := log.FromContext(r.Ctx)
+	r.reports = cmap.New[console.VulnerabilityReportAttributes]()
+	err := helpers.BackgroundPollUntilContextCancel(r.Ctx, 1*time.Minute, false, true, func(_ context.Context) (done bool, err error) {
```
Comment (michaeljguarino) [inline internal/controller/vulnerabilityreports_controller.go:172]: make this poll interval 10 minutes i think

Comment (zreigz) [reply internal/controller/vulnerabilityreports_controller.go]: done

**Pull request merged**